cluster-autoscaler/provisioningrequest/service/v1beta1client/client_test.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1client
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqwrapper"
+)
+
+func TestFetchPodTemplates(t *testing.T) {
```

Comment:
This method does not really test the real implementation, only the fake client. The only real method tested is `newAllPRsLister`, but not `NewProvisioningRequestClient`. What is the aim here?

Reply:
The `FetchPodTemplates` func is called on the `ProvisioningRequestClient` type, which is not faked.

All the dependencies are faked, but actual function is being tested here.